src/main/java/uk/gov/hmcts/darts/transcriptions/service/impl/AdminTranscriptionServiceImpl.java

```diff
@@ -161,6 +168,9 @@ public GetTranscriptionDocumentByIdResponse getTranscriptionDocumentById(Integer
     @Transactional
     public TranscriptionDocumentHideResponse hideOrShowTranscriptionDocumentById(Integer transcriptionDocumentId,
                                                                                  TranscriptionDocumentHideRequest transcriptionDocumentHideRequest) {
+        if (!this.isManualDeletionEnabled()) {
+            throw new DartsApiException(DartsApiException.DartsApiErrorCommon.FEATURE_FLAG_NOT_ENABLED);
+        }
```

Comment:
The check here needs to be more subtle. The `reason_id` in the payload needs to be checked against the `marked_for_deletion` flag on the `object_hidden_reason` table and if it's true then the flag must be enabled to proceed.

This is currently handled in the `TranscriptionDocumentHideOrShowValidator.validate` method, please look there and adjust appropriately.

Reply:
Updated